### PR TITLE
Soru/mocha

### PR DESCRIPTION
**Pull request by Sorunome**
Updates mocha

```
sorunome@sorunome-desktop repos/matrix-appservice-discord $ npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
found 0 vulnerabilities
 in 1182 scanned packages
```

adds `why-is-node-running` because newer mocha-version doesn't auto-quit, and then you can easily test for floating promises

async/await'ifies all tests

fix code & tests where floating promises existed

### CODE REVIEW

**Comment by Half-Shot** [comment]
Do we need chai-as-promised if all the tests are async now?

**Comment by Sorunome** [comment]
> Do we need chai-as-promised if all the tests are async now?

It turns out soru couldn't find out how to async-ify the last test of the user synchroniser, so that currently still uses chai-as-promised :/

File: test/config.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+const logger = new Log("MessageProcessor");
 
 // we are a test file and thus need those
 /* tslint:disable:no-unused-expression max-file-line-count */
 
 if (!argv.includes("--noisy")) {
     Log.ForceSilent();
 }
+
+after(() => {
```
Comment (Half-Shot) [inline test/config.ts]: Wait `after(()=>)` waits for the tests to all finish and then run things, that's pretty cool :o. Would this fire every time? I'm wondering if we should have some accompanying logging.

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,7 @@ logs
 *.log.*
 npm-debug.log*
 .audit.json
+*-audit.json
```
Comment (Half-Shot) [inline .gitignore]: huh?

File: test/config.ts (hunk 1 above)
Comment (Sorunome) [reply test/config.ts]: it fires once after all tests are done!

File: .gitignore (hunk 2 above)
Comment (Sorunome) [reply .gitignore]: it appears npm audit log filename patterns changed with some version of npm

Comment (Half-Shot) [reply .gitignore]: oh lovely

**Pull request merged**